### PR TITLE
Move instruction creation to core

### PR DESCRIPTION
**Pull request by bsudekum**
_Taking a 3rd stab at https://github.com/mapbox/mapbox-navigation-ios/pull/453_

This moves voice and instructions creation to core. This will allow developers to who use core to create instructions to be used in their own speech synthesizer or use the one built into ios.

todo:

- [x] move speech strings over to core
- [x] double check what must be made public
- [x] document new public methods

/cc @1ec5 @frederoni @ericrwolfe

### CODE REVIEW

File: MapboxCoreNavigation/BundleAdditions.swift [hunk 1]
```diff
@@ -28,4 +28,8 @@ extension Bundle {
             return object(forInfoDictionaryKey: "NSLocationAlwaysUsageDescription") as? String
         }
     }
+    
+    class var mapboxCoreNavigation: Bundle {
+        get { return Bundle(for: self) }
```
Comment (bsudekum) [inline MapboxCoreNavigation/BundleAdditions.swift]: Unsure what to make this value.

File: MapboxCoreNavigation/SpokenInstructionFormatter.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import Foundation
```
Comment (bsudekum) [inline MapboxCoreNavigation/SpokenInstructionFormatter.swift]: Can this file be merged with any others?

File: MapboxCoreNavigation/BundleAdditions.swift (hunk 1 above)
Comment (frederoni) [inline MapboxCoreNavigation/BundleAdditions.swift]: `mapboxCoreNavigation` makes sense. `Bundle(for: self)` should work even though it takes a class but we're inside an extension. How about the root of core navigation, which is RouteController? `Bundle(for: RouteController.self)`

File: MapboxCoreNavigation/SpokenInstructionFormatter.swift (hunk 2 above)
Comment (frederoni) [inline MapboxCoreNavigation/SpokenInstructionFormatter.swift]: I'd probably assume that this class extends NSFormatter but it doesn't need any of its functionality. I'd keep it as it is.

**Comment by bsudekum** [comment]
I also added UI instruction creation to this PR.

**Pull request merged**